src/utils/convertors/requestConverter.ts

```diff
@@ -1,8 +1,47 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { GeoServerCreateDataStoreRequest, GeoServerUpdateDataStoreRequest } from '../../common/geoserver/models/dataStore';
 import { GeoserverWorkspaceRequest } from '../../common/geoserver/models/workspace';
+import { ConnectionParams, DataStoreNameRequest } from '../../common/interfaces';
 
 /* This file contains functions that converts inputs from the geoserver-api to the request input the GeoServer itself expects to receive */
 export const workspaceRequestConverter = (workspaceName: string): GeoserverWorkspaceRequest => {
   return {
     workspace: { name: workspaceName },
   };
 };
+
+export const postDataStoreRequestConverter = (request: DataStoreNameRequest, connectionParams: ConnectionParams): GeoServerCreateDataStoreRequest => {
+  const connectionParameters = [
+    { '@key': 'host', $: connectionParams.host },
+    { '@key': 'port', $: connectionParams.port.toString() },
+    { '@key': 'database', $: connectionParams.dbName },
+    { '@key': 'user', $: connectionParams.username },
+    { '@key': 'dbtype', $: connectionParams.dbType },
+    { '@key': 'schema', $: connectionParams.schema },
+    { '@key': 'SSL mode', $: connectionParams.sslMode },
+    { '@key': 'validate connections', $: 'true' },
+  ];
+
+  if (connectionParams.password != null && connectionParams.password.trim() !== '') {
```

Comment:
check if you can use only if (connectionParams.password)

Reply:
cant, eslint warninng `  25:7  warning  Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly  @typescript-eslint/strict-boolean-expressions`